pkg/command/root.go

```diff
@@ -909,3 +926,10 @@ func printShortDescOfCmdInActiveHelp(cmd *cobra.Command, args []string) {
 		}
 	}
 }
+
+// NewRootCmdForTest creates a new instance of the root command for unit test purpose
+// Note: This must not be used as part of the production code and only used for unit tests
+func NewRootCmdForTest() (*cobra.Command, error) {
```

Comment:
I think this is where the locking should be done to allow running tests in parallel.
Without it, it is possible that one test does the `globalRootCmd = nil` and then gets scheduled out, letting another test call `NewRootCmd()`, which will change `globalRootCmd` to not nil.

If we add the lock here, please add a comment saying this is to allow to run unit tests in parallel.

Reply:
Decided to use `createRootCmd` directly that will by pass the usage of `globalRootCmd` entirely and always create a new instance for testing.